src/flow/max-flow-min-cut.jl

```diff
@@ -0,0 +1,134 @@
+"""Computes the maximum flow between the source and target vertexes in a flow
+graph using [Edmong Karp's](https://en.wikipedia.org/wiki/Edmonds%E2%80%93Karp_algorithm)
+algorithm. Returns the value of the maximum flow as well as the final flow matrix.
+
+Use a default capacity of 1 when the capacity matrix isn't specified
+"""
+
+function maximum_flow{T<:Number}(
+    flow_graph::LightGraphs.DiGraph,       # the input graph
+    source::Int,                           # the source vertex
+    target::Int,                           # the target vertex
+    capacity_matrix::AbstractArray{T,2}=   # edge flow capacities
+        DefaultDistance()
+    )
+    n = size(flow_graph)[1]                # number of vertexes
+    flow = 0
+    flow_matrix = zeros(T, n, n)           # initialize flow matrix
+
+    while true
+        v, P, S = fetch_path(flow_graph, source, target, flow_matrix, capacity_matrix)
+
+        if P == None                       # no more valid paths
+            break
```

Comment:
I'm not sure I've seen `None` used in a pythonesque way in julia - perhaps we should be using a `-1` or something here? `None` is also apparently deprecated in 0.4.